packages/flutter/test/widget/modal_barrier_test.dart

```diff
@@ -0,0 +1,109 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:test/test.dart';
+
+void main() {
+  bool tapped;
+  Widget tapTarget;
+
+  setUp(() {
+    tapped = false;
+    tapTarget = new GestureDetector(
+      child: new SizedBox(
+        width: 10.0,
+        height: 10.0,
+        child: new Text('Behind barrier')
+      ),
+      onTap: () {
+        tapped = true;
+      }
+    );
+  });
+
+  test('ModalBarrier prevents interactions with widgets behind it', () {
+    testWidgets((WidgetTester tester) {
+      Widget subject = new Stack(
+        children: [
+          tapTarget,
+          new ModalBarrier(dismissable: false),
+        ]
+      );
+
+      tester.pumpWidget(subject);
+      tester.tap(tester.findText('Behind barrier'));
+      expect(tapped, isFalse,
+        reason: 'because the tap is prevented by ModalBarrier');
+    });
+  });
+
+  test('ModalBarrier does not prevent interactions with widgets in front of it',
+      () {
```

Comment:
This indenting makes the closing brace look like it's at the wrong depth. Just exceed 80 chars here to avoid confusing integration.